go/arrow/csv/reader.go

```diff
@@ -731,15 +731,19 @@ func (r *Reader) parseList(field array.Builder, str string) {
 		field.AppendNull()
 		return
 	}
-	if !(strings.HasPrefix(str, "{") || strings.HasSuffix(str, "}")) {
+	if !(strings.HasPrefix(str, "{") && strings.HasSuffix(str, "}")) {
 		r.err = errors.New("invalid list format. should start with '{' and end with '}'")
-		field.AppendNull()
 		return
 	}
 	str = strings.TrimPrefix(str, "{")
 	str = strings.TrimSuffix(str, "}")
```

Comment:
I think this can be condensed into a single call to `strings.Trim(str, "{}")`

Reply:
I can change this but I feel the explicit way is clearer and potentially more efficient. But let me know if you want it the other way.

Reply:
Looking at the implementations from the `strings` package, I'm pretty sure that calling `strings.Trim` is more efficient than calling `TrimPrefix` and `TrimSuffix` separately. At minimum it reduces the number of code branches that get used. Personally I'd prefer using `strings.Trim` over the separate calls to prefix/suffix.

Reply:
Updated